presto-raptor/src/main/java/com/facebook/presto/raptor/systemtables/ShardMetadataRecordCursor.java

```diff
@@ -328,6 +330,26 @@ else if (temporalType.equals(TIMESTAMP)) {
         return builder.build();
     }
 
+    private static void buildPredicatesAndValues(Domain nameDomain, List<String> values, List<String> predicates, String predicatePrefix)
+    {
+        if (nameDomain.isSingleValue()) {
```

Comment:
Do we need this special case? I think the range code below will work for both cases. It's fine if we generate an in-list with only one value: `xxx IN (?)`

Reply:
Thanks for pointing this out. I love simpler codes.